### PR TITLE
fix(action): the build directory_md workflow

### PR DESCRIPTION
**Pull request by imp2002**
1. use a single `.py` file maybe better than write it in `.yml`.
2. fix the script regard mod.rs as a entry.

### CODE REVIEW

File: DIRECTORY.md [hunk 1]
```diff
@@ -21,28 +19,24 @@
...
     * [Binary Search](https://github.com/TheAlgorithms/Rust/blob/master/src/searching/binary_search.rs)
     * [Linear Search](https://github.com/TheAlgorithms/Rust/blob/master/src/searching/linear_search.rs)
-    * [Mod](https://github.com/TheAlgorithms/Rust/blob/master/src/searching/mod.rs)
   * Sorting
     * [Bubble Sort](https://github.com/TheAlgorithms/Rust/blob/master/src/sorting/bubble_sort.rs)
     * [Counting Sort](https://github.com/TheAlgorithms/Rust/blob/master/src/sorting/counting_sort.rs)
     * [Heap Sort](https://github.com/TheAlgorithms/Rust/blob/master/src/sorting/heap_sort.rs)
     * [Insertion Sort](https://github.com/TheAlgorithms/Rust/blob/master/src/sorting/insertion_sort.rs)
     * [Merge Sort](https://github.com/TheAlgorithms/Rust/blob/master/src/sorting/merge_sort.rs)
-    * [Mod](https://github.com/TheAlgorithms/Rust/blob/master/src/sorting/mod.rs)
```
Comment (Panquesito7) [inline DIRECTORY.md:41]: Why is this being deleted?

**Review by Panquesito7 — COMMENTED** [review]
I don't know Python, but overall it looks good. 🙂👍

File: DIRECTORY.md (hunk 1 above)
Comment (imp2002) [reply DIRECTORY.md:41]: mod.rs is to combine modules of different levels, not a algorithm, maybe should't regard as a entry in algorithm directory.😊

**Review by Panquesito7 — DISMISSED** [review]
Thank you, @imp2002, for your contribution. 😄👍
I'll wait for @siriak to review this PR before merging.

**Review by siriak — DISMISSED** [review]
LGTM, thanks!

**Review by siriak — APPROVED** [review]

**Pull request merged**